### PR TITLE
xcube-758: xcube server fails to resolve datasets for relative parent paths

### PR DESCRIPTION
**Pull request by TonioF**
Solves #758 

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* ~[ ] Add docstrings and API docs for any new/modified user-facing classes and functions~
* ~[ ] New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Review by forman — COMMENTED** [review]
Please consider my comments. Otherwise, fine!

File: xcube/webapi/datasets/context.py [hunk 1]
```diff
@@ -842,18 +818,52 @@ def _is_not_empty(prefix):
...
+
+
 def _get_common_prefixes(p):
     # Recursively examine a list of paths for common prefixes:
     # If no common prefix is found, split the list in half and
     # examine each half separately
     prefix = os.path.commonprefix(p)
+    # ensure prefix does not end with full or partial directory
+    # or file name
+    prefix = prefix[:max(_lastindex(prefix, '/'), _lastindex(prefix, '\\'), 0)]
```
[the remark this answers is not in the document]
Comment (TonioF) [reply xcube/webapi/datasets/context.py:852]: Then the prefix will not be changed.

File: xcube/webapi/datasets/context.py [hunk 2]
```diff
@@ -842,18 +818,52 @@ def _is_not_empty(prefix):
...
     if _is_not_empty(prefix) or len(p) == 1:
         return [prefix]
     else:
         return _get_common_prefixes(p[:int(len(p) / 2)]) + \
                _get_common_prefixes(p[int(len(p) / 2):])
 
 
+def _lastindex(prefix, symbol):
+    try:
+        return len(prefix) - prefix[::-1].index(symbol) - 1
```
[the remark this answers is not in the document]
Comment (TonioF) [reply xcube/webapi/datasets/context.py]: It's used now.

File: xcube/webapi/datasets/context.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (forman) [reply xcube/webapi/datasets/context.py:852]: > Then the prefix will not be changed.

But you cannot index an empty string:

```
>>> ''[0]
Traceback (most recent call last):
...
IndexError: string index out of range
```

File: xcube/webapi/datasets/context.py [hunk 3]
```diff
@@ -842,18 +818,52 @@ def _is_not_empty(prefix):
...
                _get_common_prefixes(p[int(len(p) / 2):])
 
 
+def _lastindex(prefix, symbol):
+    try:
+        return prefix.rindex(symbol)
+    except ValueError:
+        return -1
+
+
```
Comment (forman) [inline xcube/webapi/datasets/context.py:866]: Consider replacing the while function by `prefix.rfind(symbol)`. (https://docs.python.org/3/library/stdtypes.html#str.rfind)

**Review by forman — CHANGES_REQUESTED** [review]

File: xcube/webapi/datasets/context.py (hunk 3 above)
Comment (TonioF) [reply xcube/webapi/datasets/context.py:866]: You cannot ask for a slice:

```
>>> ''[0]
''
```

Comment (TonioF) [reply xcube/webapi/datasets/context.py:866]: Used rfind

File: xcube/webapi/datasets/context.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (forman) [reply xcube/webapi/datasets/context.py:852]: oops, overlooked the `:`.

File: xcube/webapi/datasets/context.py (hunk 3 above)
Comment (forman) [reply xcube/webapi/datasets/context.py:866]: > You can ask for a slice:
> 
> ```
> >>> ''[0]
> ''
> ```

Almost
 ```
>>> ''[:0]
''
```

**Review by forman — APPROVED** [review]

**Comment by forman** [comment]
@TonioF note that "Solves #758" does not auto-close the issue. Use "Closes #758" instead.

**Pull request merged**